src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/mapping/MarkerMapper.kt

```diff
@@ -13,10 +13,10 @@ class MarkerMapper {
         )
     }
 
-    fun mapToEntity(type: Marker): MarkerEntity {
+    fun mapToEntity(type: Marker, takeFk: Int? = null): MarkerEntity {
```

Comment:
I'd prefer to have the actual take object passed in, and then use take.id

Reply:
There are instances where we have the takeFk, but not the take object itself to pass in. In those instances I think we have the TakeEntity, so we could pass in the TakeEntity instead of just the id, if that would be better.

Reply:
can you point me to such instances?

Reply:
Update method of TakeRepository, CollectionRepository, and ChunkRepository. We fetch the entity so we don't overwrite the existing relationship data, but we never have the full object

Reply:
I suppose these are just more internal private methods anyway